lib/libzfs/libzfs_import.c

```diff
@@ -148,23 +148,71 @@ zpool_clear_label(int fd)
 	int l;
 	vdev_label_t *label;
 	uint64_t size;
+	int labels_cleared = 0;
 
 	if (fstat64_blk(fd, &statbuf) == -1)
 		return (0);
+
 	size = P2ALIGN_TYPED(statbuf.st_size, sizeof (vdev_label_t), uint64_t);
 
 	if ((label = calloc(1, sizeof (vdev_label_t))) == NULL)
 		return (-1);
 
 	for (l = 0; l < VDEV_LABELS; l++) {
-		if (pwrite64(fd, label, sizeof (vdev_label_t),
+		uint64_t state, guid;
+		nvlist_t *config;
+
+		if (pread64(fd, label, sizeof (vdev_label_t),
 		    label_offset(size, l)) != sizeof (vdev_label_t)) {
-			free(label);
-			return (-1);
+			continue;
+		}
+
+		if (nvlist_unpack(label->vl_vdev_phys.vp_nvlist,
+		    sizeof (label->vl_vdev_phys.vp_nvlist), &config, 0) != 0) {
+			continue;
+		}
+
+		if ((nvlist_lookup_uint64(config, ZPOOL_CONFIG_GUID,
+		    &guid) != 0) != 0) {
```

Comment:
You only need `!= 0` check

Reply:
Fixed.  I also took the opportunity to pull the `guid == 0` check up in to this conditional to be consistent with the following `state` check.